Feat:Add Meetup History

ritwickraj78 opened the pull request:
### Description
 - Added Images Model and Resource Fields to Add Meetup and Edit Meetup Form
- Created past meetup form where user can upload resources and images after meetup is held.
- Created Past Meetup View to add the resources.

Fixes #548 

### Type of Change:

- Code


**Code/Quality Assurance Only**

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

- All unit tests pass
- Flake test passes
- Travis Check successful.
- Tested locally

![Screenshot from 2020-07-10 11-49-38](https://user-images.githubusercontent.com/43514828/87123797-12431500-c2a5-11ea-8684-47f63881fd38.png)
![Screenshot from 2020-07-10 11-49-35](https://user-images.githubusercontent.com/43514828/87123802-14a56f00-c2a5-11ea-90f9-9fbfa1790dbf.png)
![Screenshot from 2020-07-10 11-45-49](https://user-images.githubusercontent.com/43514828/87123806-166f3280-c2a5-11ea-8aed-e29c625adc2f.png)
![Screenshot from 2020-07-10 11-45-23](https://user-images.githubusercontent.com/43514828/87123807-1707c900-c2a5-11ea-9193-902ebf876866.png)
![Screenshot from 2020-07-10 11-45-02](https://user-images.githubusercontent.com/43514828/87123810-1838f600-c2a5-11ea-8995-b4aaae79f932.png)


### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

ritwickraj78 committed:
Meetup History Backend Added

SanketDG commented on systers_portal/meetup/forms.py:
```diff
@@ -63,12 +66,14 @@ class AddMeetupForm(ModelFormWithHelper):
         helper_class = SubmitCancelFormHelper
         helper_cancel_href = "{% url 'index' %}"
 
+    images = MultiFileField(required=False)
```
Apologies if you have already gone through this

But why can't you just use https://docs.djangoproject.com/en/3.0/topics/http/file-uploads/#uploading-multiple-files

ritwickraj78 replied:
Using the above actuallly requires changing the views.py which was too much work, using the way the library works, it was much easier to do it without changing the views or the forms much. Because the way the views are written rn, it becomes really very difficult to make even a minor change to them

SanketDG replied:
Got you!

SanketDG commented on requirements/dev.txt:
```diff
@@ -13,3 +13,6 @@ selenium==3.12.0
 pinax-notifications==5.0.3
+geoip2==3.0.0
+django-ipware==3
+django-multiupload
```
please pin a specific version

SanketDG replied:
Also you do not need add these to both of the files, because dev inherits from prod here, so just keep adding in prod, make sure you update for the other PR too.

ritwickraj78 replied:
Sure!

SanketDG commented on systers_portal/meetup/views.py:
```diff
@@ -821,3 +821,32 @@ def post(self, request):
+    model = Meetup
+    slug_url_kwarg = "meetup_slug"
+    form_class = PastMeetup
+    form_valid_message = (u"Meetup edited Successfully")
```
Let's change this to something else like "Images have been uploaded"

ritwickraj78 replied:
Will do that. Thanks for catching this

SanketDG reviewed (changes requested):
Initial set of reviews. 

Also, if we are not planning to have "videos" as a resource anytime soon, let's just name everything as `Image` instead of `Resource`

ritwickraj78 committed:
Meetup History Frontend Added

ritwickraj78 commented:
@SanketDG @sakshi1499 @sammy1997 Instead of opening two PRs for front end and backend, I have made two different commits as the two are quite interrelated and would make more sense in one PR. Please Review. @SanketDG Changes you suggested have been kept in mind

sakshi1499 commented on systers_portal/meetup/forms.py:
```diff
@@ -104,14 +112,25 @@ def clean_time(self):
     class Meta:
         model = Meetup
-        fields = ('title', 'slug', 'date', 'time', 'description', 'venue')
+        fields = ('title', 'slug', 'date', 'time', 'description', 'venue',)
```
Why have you added an extra comma?

ritwickraj78 replied:
I was trying out different ways so left the comma there. Will remove it :)

sakshi1499 commented on systers_portal/templates/meetup/meetup.html:
```diff
@@ -1,63 +1,73 @@
+    <thead>
+      <tr>
+        <th>Coming</th>
+        <th>Not Coming</th>
```
Why are we showing the number of people not coming?

ritwickraj78 replied:
Even I am not aware. I will remove it

sakshi1499 replied:
Okay cool

sakshi1499 reviewed (commented):
I haven't tested this locally yet. Will do it asap

sammy1997 reviewed (commented):
.

sammy1997 commented:
So many commits!! @ritwickraj78 if the work is independent of the previous PRs, decouple it and make the branch off develop

ritwickraj78 committed:
Code: Dropdown Menu and Meetup search Revamped

* Meetup Search Revamped

* Dropdown Revamp

Meetup Search Revamped

Added Ipware and Scripts

Dropdown Revamp

Added Image & Resource Field

SanketDG reviewed (commented):
Looks good from a eye through. The rebase has not happened yet (I still see old commits), if you somehow messed up the older commits, please squash it with the first new commit, that should not involve much work.

Will test locally soon.

ritwickraj78 commented:
> Looks good from a eye through. The rebase has not happened yet (I still see old commits), if you somehow messed up the older commits, please squash it with the first new commit, that should not involve much work.
> 
> Will test locally soon.

I have rebased it. Will have to squash the commits
